drivers/s3.go

```diff
@@ -102,13 +102,10 @@ func customS3Region(host string) string {
 	}
 	hostname := u.Hostname()
 	parts := strings.Split(hostname, ".")
-	if len(parts) == 1 {
-		return hostname
+	if len(parts) >= 4 {
+		return parts[1]
 	}
-	if len(parts) == 2 {
-		return parts[0]
-	}
-	return parts[1]
+	return defaultIgnoredRegion
```

Comment:
This whole code (in general, not your change) makes me so nervous, but hopefully we have good enough unit test coverage by this point to be reasonable confident in it

Reply:
Yes, I don't like, it because we make some "assumptions" about how the `endpoint` URL looks like. I think that a better way would be to maybe always include the `region` in our `s3+https://` URL. Though, probably too many changes to do it right now.